### PR TITLE
Fix imports when some optional dependencies are not installed

### PR DESCRIPTION
**Pull request by stellasia**
# Description

ImportError were raised if numpy or openai where not installed, when trying to import another embedder/llm.

## Type of Change
- [ ] New feature
- [x] Bug fix
- [ ] Breaking change
- [ ] Documentation update
- [ ] Project configuration change

## Complexity

> **Note**
>
> Please provide an estimated complexity of this PR of either Low, Medium or High
>
>

Complexity:

## How Has This Been Tested?
- [x] Unit tests
- [x] E2E tests
- [x] Manual tests

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [ ] Documentation has been updated
- [x] Unit tests have been updated
- [x] E2E tests have been updated
- [ ] Examples have been updated
- [ ] New files have copyright header
- [ ] CLA (https://neo4j.com/developer/cla/) has been signed
- [x] CHANGELOG.md updated if appropriate

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## Next
 
+### Fixed
+- Fix a bug where `openai` Python client and `numpy` were required to import any embedder or LLM.
```
Comment (willtai) [inline CHANGELOG.md:6]: Shall we have the alpha-releases in the CHANGELOG or we can put everything in the next release?

**Comment by letoribo** [comment]
I was forced to create this fork to fix my Vercel deployment:
https://github.com/letoribo/neo4j-graphrag-python/commit/a07e4682f5196a1191a887a9976ba27a403fa6d3

it would be good to solve this problem when the developer does not use OpenAI, AzureOpenAI or SentenceTransformer

File: src/neo4j_graphrag/embeddings/openai.py [hunk 2]
```diff
@@ -25,26 +26,29 @@
     openai = None  # type: ignore
 
 
-class OpenAIEmbeddings(Embedder):
+class BaseOpenAIEmbeddings(Embedder, abc.ABC):
```
Comment (willtai) [inline src/neo4j_graphrag/embeddings/openai.py:29]: Why have this as a separate base class? Is this too much indirection?

Comment (stellasia) [reply src/neo4j_graphrag/embeddings/openai.py:29]: What would you suggest instead?

File: CHANGELOG.md (hunk 1 above)
Comment (stellasia) [reply CHANGELOG.md:6]: We have the alpha release in the Changelog, right? The only thing is that we'll need to copy all the changes under Next before releasing the final version.

**Comment by stellasia** [comment]
@letoribo thank you, this is the bug this PR is fixing. Feel free to open issues in the future if you face other pain point.

**Review by willtai — APPROVED** [review]

**Pull request merged**